apps/ws/src/socket/ws.gateway.ts

```diff
@@ -28,7 +29,12 @@ export class WSGateway implements OnGatewayConnection {
       return this.disconnect(connection);
     }
 
-    return await connection.join(subscriber._id);
+    await connection.join(subscriber._id);
+    await this.handleOnlineStatus(subscriber, true);
+
+    connection.on('disconnect', async () => {
+      await this.handleOnlineStatus(subscriber, false);
+    });
```

Comment:
hey @BiswaViraj I'm missing here the logic that is described in the ticket, do you plan to add it in the separate PR?

Reply:
Had some issues with the multi-tab situation, initially intended to do it with a diff PR, will include it in this PR now.